### ISSUE TITLE
Multi-instance user_nl files not created correctly for NCK tests

### ISSUE THREAD

**Issue by billsacks** [ESMCI/cime#469]
When I create an NCK test like `NCK_D_Ld3.f10_f10.ICLM45.yellowstone_gnu.clm-default`, there is a user_nl_clm file that has lots of important stuff. However, the user_nl_clm_0001 and user_nl_clm_0002 files just contain the blank templates that are initially created for user_nl_clm files:

```
!----------------------------------------------------------------------------------
! Users should add all user specific namelist changes below in the form of
! namelist_var = new_namelist_value
!
! EXCEPTIONS:
! Set use_cndv           by the compset you use and the CLM_BLDNML_OPTS -dynamic_vegetation setting
! Set use_vichydro       by the compset you use and the CLM_BLDNML_OPTS -vichydro           setting
! Set use_cn             by the compset you use and CLM_BLDNML_OPTS -bgc  setting
! Set use_crop           by the compset you use and CLM_BLDNML_OPTS -crop setting
! Set spinup_state       by the CLM_BLDNML_OPTS -bgc_spinup      setting
! Set irrigate           by the CLM_BLDNML_OPTS -irrig           setting
! Set co2_ppmv           with CCSM_CO2_PPMV                      option
! Set dtime              with L_NCPL                             option
! Set fatmlndfrc         with LND_DOMAIN_PATH/LND_DOMAIN_FILE    options
! Set finidat            with RUN_REFCASE/RUN_REFDATE/RUN_REFTOD options for hybrid or branch cases
!                        (includes $inst_string for multi-ensemble cases)
! Set glc_smb            with GLC_SMB                            option
! Set maxpatch_glcmec    with GLC_NEC                            option
! Set glc_do_dynglacier  with GLC_TWO_WAY_COUPLING               env variable
!----------------------------------------------------------------------------------
```

It appears the logic for copying user_nl_* to the multi-instance user_nl files is correct for _N2 tests, but not for NCK tests.

**Comment by billsacks** [comment]
@jedwards4b can you please look at this?

**Comment by billsacks** [comment]
At first I thought the problem could be fixed by creating an nck test with the new SystemTestsCompareTwo infrastructure, but that turns out to have this same problem. Nevertheless, I'll submit a PR with the new NCK test, because it seems more maintainable.

**Comment by billsacks** [comment]
I realized that one solution to this could be to have some special-purpose logic in the nck test to do the necessary copies. (I'm thinking that the general-purpose code may not be able to handle this because at the point of test creation, it thinks there's going to be a single instance.)

**Comment by billsacks** [comment]
> I realized that one solution to this could be to have some special-purpose logic in the nck test to do the necessary copies.

Here's some pseudocode for what I was thinking about:

```
for each component 'foo' in [get component list]:
    if user_nl_foo exists:
        if user_nl_foo_0001 doesn't exist: copy user_nl_foo to user_nl_foo_0001
        if user_nl_foo_0002 doesn't exist: copy user_nl_foo to user_nl_foo_0002
```

**Comment by jedwards4b** [comment]
fixed by #483
